babel/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -692,3 +692,80 @@ SqlNode DateTimeTerm() :
         )
     )
 }
+
+// This excludes CompoundIdentifier() as a data type.
+SqlDataTypeSpec DataTypeAlternativeCastSyntax() :
```

Comment:
If we go with the recommendation below, this function can just be removed and the existing DataType() function in Parser.jj can also have a boolean parameter for whether compound identifiers are allowed.